### PR TITLE
[IMP] website_*: fine-tune actions buttons design

### PR DESCRIPTION
**Pull request by chgo-odoo**
*: sale, sale_wishlist, sale_comparison_wishlist, sale_comparison

### Main goal
This PR fine-tunes the design of the action buttons (add to compare, remove from wishlist, remove from recently viewed) in the following places:

- Dynamic product snippets (with the recently viewed option)
- Shop page
- Wishlist page

Previously, these buttons used a temporary design. They are now adjusted to match the overall style of /shop.

To improve consistency, we:

- Apply the “glass effect” style in grid mode to match other buttons
- Position the button on the far right in list view, or in the top corner in grid view

### Additional fixes included :
- Correcting minor alignment issues in button positioning on both shop and wishlist pages
- Rendering the comparison placeholder, already present on the shop page, on the wishlist page as well
- Aligning certain design elements of the wishlist page with those of the shop page for consistency

task-4997342
part of task-4252024

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_comparison_wishlist/views/templates.xml [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-                <small><i t-attf-class="fa fa-exchange small"></i></small>
-            </button>
+
+            <t t-if="is_view_active('website_sale_comparison.add_to_compare')">
+                <button
+                    t-if="product_variant_id and categories"
+                    type="button"
+                    class="btn btn-light o_add_to_compare"
+                    t-att-data-product-id='wish.product_id.id'>
+                    <i class="fa fa-fw fa-exchange"/>
```
Comment (mano-odoo) [inline addons/website_sale_comparison_wishlist/views/templates.xml]:
```suggestion
                    <i class="fa fa-fw fa-exchange" role="img"/>
```

File: addons/website_sale_comparison_wishlist/views/templates.xml [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+                </button>
+
+                <!-- Render placeholder when product is not comparable to ensure alignment -->
+                <t t-else="">
+                    <div class="o_add_to_compare_placeholder pe-none opacity-0 btn btn-light">
+                        <span class="fa fa-exchange fa-fw"/>
```
Comment (mano-odoo) [inline addons/website_sale_comparison_wishlist/views/templates.xml]:
```suggestion
                        <i class="fa fa-exchange fa-fw" role="img"/>
```

File: addons/website_sale_comparison/static/src/scss/website_sale_comparison.options.scss [hunk 3]
```diff
@@ -23,6 +23,12 @@
     }
 }
 
+// Wishlist comparison button placeholder - always visible when comparison module is active
+.o_add_to_compare_placeholder {
+    display: inline-flex;
+    order: var(--o-wsale-comparison-btn-order);
+}
```
Comment (mano-odoo) [inline addons/website_sale_comparison/static/src/scss/website_sale_comparison.options.scss]: Can't we name the class on the element `o_add_compare_placeholder` and benefit from the already existing styling? Applying `d-inline-flex` as a utility class.

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml [hunk 4]
```diff
@@ -375,6 +366,15 @@
                                                 </button>
                                             </div>
                                         </t>
+                                        <t t-set="extra_actions">
+                                            <button t-if="wishes and wish.product_id.product_tmpl_id == product"
+                                                type="button"
+                                                class="btn btn-light o_wish_rm rounded-pill lh-1"
+                                                title="Remove from wishlist"
+                                            >
+                                                <i class="oi oi-close" role="presentation"/>
```
Comment (mano-odoo) [inline addons/website_sale_wishlist/views/website_sale_wishlist_template.xml:386]:
```suggestion
                                                <i class="oi oi-close" role="img"/>
```

File: addons/website_sale/static/src/scss/product_tile.scss [hunk 5]
```diff
@@ -350,20 +350,21 @@
         display: var(--o-wsale-card-btn-label-display);
     }
     .o_carousel_product_remove {
-        @include o-position-absolute(7%, 50%);
-        transform: translateX(50%);
-        display: inline-flex;
+        --btn-padding-x: .7rem; // To match the default value from on hover buttons option
```
Comment (mano-odoo) [inline addons/website_sale/static/src/scss/product_tile.scss]: If we replicate the padding to match the size of the other hover buttons, there is still 4px difference between the add to cart and the remove. We should probably replicate the same size.

I notice as well that we're not doing the same thing in the wishlist, so maybe it's better to stay consistent. I think smaller remove buttons are fine (just make sure they are usable on touch devices)

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss [hunk 6]
```diff
@@ -35,19 +35,76 @@
...
+
+    --o-wsale-wishlist-card-offset-top: calc(var(--o-wsale-card-offset-top) + var(--btn-padding-x));
+    --o-wsale-wishlist-card-offset-x: calc(var(--o-wsale-card-offset-x) + var(--btn-padding-x));
+
+    --o-wsale-wishlist-btn-offset-top-default: calc(var(--o-wsale-card-border-radius, 0px) * 0.1 + var(--btn-padding-y));
+    --o-wsale-wishlist-btn-offset-end-default: calc(var(--o-wsale-card-border-radius, 0px) * 0.2 + var(--btn-padding-x));
+
+    @include o-position-absolute(
+        $top: var(--o-wsale-wishlist-card-offset-top, var(--o-wsale-wishlist-btn-offset-top-default)),
+        $right: var(--o-wsale-wishlist-card-offset-x, var(--o-wsale-wishlist-btn-offset-end-default))
```
Comment (mano-odoo) [inline addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss:82]: Why do we need fallback values, `--o-wsale-wishlist-card-offset-top` and `--o-wsale-wishlist-card-offset-x` are defined in the same scope, is there any place where the fallback value applies?

If it can be removed, check other occurences

File: addons/website_sale/static/src/scss/product_tile.scss [hunk 7]
```diff
@@ -350,20 +350,21 @@
...
+
+        @include o-position-absolute(
+            $top: var(--o-wsale-wishlist-card-offset-top, var(--o-wsale-wishlist-btn-offset-top-default)),
+            $right: var(--o-wsale-wishlist-card-offset-x, var(--o-wsale-wishlist-btn-offset-end-default))
+        );
+        @include o-wsale-card-btn-design-subtle($body-bg, $body-color);
         aspect-ratio: 1;
-        align-items: center;
-        justify-content: center;
-        transition: none;
```
Comment (mano-odoo) [inline addons/website_sale/static/src/scss/product_tile.scss:374]: Nitpick the empty line should be between the property and the `@include`

File: addons/website_sale/static/src/scss/product_tile.scss [hunk 8]
```diff
@@ -590,6 +591,7 @@
     --o-wsale-card-border-width: #{$border-width};
     --o-wsale-card-border-radius: var(--o-wsale-opt-border-radius, 0px);
     --o-wsale-card-padding: var(--_padding-base);
+    --o-wsale-card-btns-row-padding: 0;
```
Comment (mano-odoo) [inline addons/website_sale/static/src/scss/product_tile.scss:624]: Just a note we will need to use the same approach in https://github.com/odoo/odoo/pull/222504/files#diff-43a467c990bdc6854b7afe622f25f47aee580aac6615784489d3cf14b39ba65aR611-R613

File: addons/website_sale/static/src/scss/product_tile.scss [hunk 9]
```diff
@@ -769,8 +773,16 @@
...
+
+            &:where(:not(.o_wsale_product_btn_primary)) {
+                --o-wsale-card-btn-padding-x: 0;
+            }
         }
     }
+
+    .o_add_to_compare {
+        display: inline-flex;
+    }
```
Comment (mano-odoo) [inline addons/website_sale/static/src/scss/product_tile.scss]: See previous comment, check if this could be an utility class rather

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss [hunk 10]
```diff
@@ -35,19 +35,76 @@
...
+    );
+    z-index: 3;
```
Comment (mano-odoo) [inline addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss]: Why is it needed? Since we apply the raw property, if it's really needed could we apply it with the `z-3` utility?

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss [hunk 11]
```diff
@@ -35,19 +35,76 @@
...
+
+        --btn-padding-x: .7rem;
 
         @include media-breakpoint-up(md) {
-            grid-column: auto/span 4;
+            position: static;
+            align-self: center;
+            margin-left: $spacer;
         }
+        order: 13;
```
Comment (mano-odoo) [inline addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss]: I suppose we use 13 because it's the value of `.order-last`, I'd suggest still using the map-get then.

Or keep track of the precedently ordered items with a custom property and re-use this CP here with a order: calc(CP+1). Or have your custom map of ordered items for this context.

```suggestion
            order: map-get($order-array, "last");
```

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss
Comment (mano-odoo) [inline addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss:1]: There are still some finetunes to be made here is what I've noticed, I put in blue things that may not be related to your changes
list_grid
showcase
thumbnails_maybe_expected_shop_does_the_same

The last one is probably not a bug/not your change since it's used on the /shop page as well, but it felt a bit weird wdyt?

**Review by mano-odoo — CHANGES_REQUESTED** [review]
Good job on this, thanks a lot! A few comments below

File: addons/website_sale_comparison_wishlist/views/templates.xml (hunk 2 above)
Comment (chgo-odoo) [reply addons/website_sale_comparison_wishlist/views/templates.xml]: Should be `aria-hidden="true"` instead IMO. We don't want screen readers to take this into account

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml (hunk 4 above)
Comment (chgo-odoo) [reply addons/website_sale_wishlist/views/website_sale_wishlist_template.xml:386]: There is a title on the button, presentation should be the one to use so it's ignored by screenreaders

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss
Comment (chgo-odoo) [reply addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss:1]: Last one could (probably) use the add to cart without label IMO, to be checked

File: addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss (hunk 6 above)
Comment (chgo-odoo) [reply addons/website_sale_wishlist/static/src/scss/website_sale_wishlist.scss:82]: It depends on the layout, in some cases `--o-wsale-card-offset-top` will be undefined and so `--o-wsale-wishlist-card-offset-top` will compute a wrong value.

This approach is already in use in `/shop` so I sticked to that

**Comment by chgo-odoo** [comment]
Hello @mano-odoo, thanks for your review, I pushed the changes in https://github.com/odoo/odoo/pull/222953/commits/f9bed781900a568a235409127c18efb05eace7f6

Thanks !

File: addons/website_sale/static/src/scss/product_tile.scss [hunk 12]
```diff
@@ -350,20 +350,22 @@
...
     }
     .o_carousel_product_remove {
-        @include o-position-absolute(7%, 50%);
-        transform: translateX(50%);
-        display: inline-flex;
-        aspect-ratio: 1;
-        align-items: center;
-        justify-content: center;
-        transition: none;
+        --btn-padding-x: #{map-get($spacers, 2)}; // To match the default value from on hover buttons option
```
Comment (mano-odoo) [inline addons/website_sale/static/src/scss/product_tile.scss]: You can remove the comment it's not relevant anymore
```suggestion
        --btn-padding-x: #{map-get($spacers, 2)};
```

**Review by mano-odoo — CHANGES_REQUESTED** [review]
Hello, thanks for your changes, don't forget to address 
https://github.com/odoo/odoo/pull/222953#discussion_r2297958056
https://github.com/odoo/odoo/pull/222953#discussion_r2297658519

The rest is good, thanks! (Don't forget the PR/Commit description)

**Review by mano-odoo — APPROVED** [review]
LGTM, don't forget the task-id in the PR description 😇

File: addons/website_sale/views/product_tile_templates.xml [hunk 13]
```diff
@@ -230,6 +230,8 @@
                     <t t-else="" t-call="website_sale.shop_product_buttons"/>
                 </div>
             </div>
+            <!-- Extra actions (wishlist remove, ...) -->
+            <t t-if="extra_actions" t-out="extra_actions"/>
```
Comment (livaios) [inline addons/website_sale/views/product_tile_templates.xml:233]: For what you are trying to achieve here its better to use an override of the template 
```suggestion
```

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml [hunk 14]
```diff
@@ -375,6 +366,15 @@
...
                                         </t>
+                                        <t t-set="extra_actions">
+                                            <button t-if="wishes and wish.product_id.product_tmpl_id == product"
+                                                type="button"
+                                                class="btn btn-light o_wish_rm z-3 rounded-pill lh-1"
+                                                title="Remove from wishlist"
+                                            >
+                                                <i class="oi oi-close" role="presentation"/>
+                                            </button>
+                                        </t>
```
Comment (livaios) [inline addons/website_sale_wishlist/views/website_sale_wishlist_template.xml]: You will need to define this template inheritance outside of this template.
```
    <template id="website_sale_wishlist.products_item" inherit_id="website_sale.products_item">
        <div name="o_product_action_buttons" class="o_wsale_product_btn" position="after">
            <button 
                t-if="wishes and wish.product_id.product_tmpl_id == product"
                type="button"
                class="btn btn-light o_wish_rm z-3 rounded-pill lh-1"
                title="Remove from wishlist"
            >
                <i class="oi oi-close" role="presentation"/>
            </button>
        </div>
    </template>
```
 I also suggest using a name instead of the class so you will need to add a name to that previous div in `website_sale`
https://github.com/odoo/odoo/blob/98581880a48d59eb395f79f9508ff2fe82d5d1db/addons/website_sale/views/product_tile_templates.xml#L228

File: addons/website_sale_comparison_wishlist/views/templates.xml [hunk 15]
```diff
@@ -1,16 +1,27 @@
...
-                <small><i t-attf-class="fa fa-exchange small"></i></small>
-            </button>
+
+            <t t-if="is_view_active('website_sale_comparison.add_to_compare')">
+                <button
+                    t-if="product_variant_id and categories"
+                    type="button"
+                    class="btn btn-light o_add_to_compare d-lg-inline-flex"
+                    t-att-data-product-id='wish.product_id.id'
+                    aria-label="Add to compare">
```
Comment (livaios) [inline addons/website_sale_comparison_wishlist/views/templates.xml]:
```suggestion
                    aria-label="Add to compare"
                >
```

**Review by livaios — CHANGES_REQUESTED** [review]
Hello @chgo-odoo 

Thank you for your work, I just have a few comments

File: addons/website_sale_comparison_wishlist/views/templates.xml [hunk 16]
```diff
@@ -1,16 +1,28 @@
...
+                >
+                    <i class="fa fa-fw fa-exchange" role="presentation"/>
+                </button>
+
+                <!-- Render placeholder when product is not comparable to ensure alignment -->
+                <t t-else="">
+                    <div class="o_add_to_compare_placeholder btn btn-light pe-none opacity-0 d-lg-inline-flex">
+                        <i class="fa fa-exchange fa-fw" aria-hidden="true"/>
+                    </div>
+                </t>
```
Comment (livaios) [inline addons/website_sale_comparison_wishlist/views/templates.xml]: in `/shop` we didn't need a placeholder and its aligned fine is there a reason why we need it here?

File: addons/website_sale_stock_wishlist/views/website_sale_stock_wishlist_templates.xml [hunk 17]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
-    <template id="product_wishlist" inherit_id="website_sale_wishlist.product_wishlist">
+    <template id="product_wishlist" inherit_id="website_sale.products_item">
```
Comment (livaios) [inline addons/website_sale_stock_wishlist/views/website_sale_stock_wishlist_templates.xml]:
```suggestion
    <template id="product_wishlist" inherit_id="website_sale_wishlist.product_wishlist">
```
You should override the product_wishlist template because you are only changing the wishlist view, if you inherit product_item it will affect all elements using product_item and it will behave differently if wishlist is installed or not

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml [hunk 18]
```diff
@@ -384,5 +388,4 @@
             </div>
         </t>
     </template>
-
```
Comment (livaios) [inline addons/website_sale_wishlist/views/website_sale_wishlist_template.xml]: unnecessary diff

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml [hunk 19]
```diff
@@ -44,6 +44,19 @@
         </t>
     </template>
 
+    <template id="website_sale_wishlist.products_item" inherit_id="website_sale.products_item">
```
Comment (livaios) [inline addons/website_sale_wishlist/views/website_sale_wishlist_template.xml]: Same comment here you are overriding the general `products_item` here you can modify it directly in the wishlist template

**Review by livaios — CHANGES_REQUESTED** [review]

File: addons/website_sale_comparison_wishlist/views/templates.xml (hunk 16 above)
Comment (chgo-odoo) [reply addons/website_sale_comparison_wishlist/views/templates.xml]: We do have one on the shop page, here is an example in the list view layout image

**Comment by chgo-odoo** [comment]
Hello @livaios 👋 

This one is finally ready 🥴 FYI I added a new change here https://github.com/odoo/odoo/compare/b7f6742e71fc36b311ecd1981d821bb0a326c92f..245a85e4d0e9725f2f10c1bcaf101491eb1107bd as there was still an issue with the buttons layout in the card list layout

Thanks

File: addons/website_sale_wishlist/views/website_sale_wishlist_template.xml [hunk 20]
```diff
@@ -382,6 +373,15 @@
                                                 </button>
                                             </div>
                                         </t>
+                                        <t t-set="extra_actions">
+                                            <button t-if="wishes and wish.product_id.product_tmpl_id == product"
```
Comment (livaios) [inline addons/website_sale_wishlist/views/website_sale_wishlist_template.xml]: We want to avoid setting "extra_actions" if we're not showing the button anyways
```suggestion
                                        <t 
                                            t-if="wish and wish.product_id.product_tmpl_id == product" 
                                            t-set="extra_actions"
                                        >
                                            <button
```

Also we can check for `t-if="wish and ..."` instead of `wishes`

**Review by livaios — CHANGES_REQUESTED** [review]
Hello @chgo-odoo,

Just one small comment and it should be good to go :slightly_smiling_face:

**Review by livaios — APPROVED** [review]
Hello @chgo-odoo,

Thanks for your work!
LGTM :+1:

**Review by chevalierv — APPROVED** [review]
Hello 

LGTM 👍 

robodoo r+

Have a nice day!

**Comment by chgo-odoo** [comment]
@chevalierv pushed to solve conflicts with https://github.com/odoo/odoo/commit/c0077eee67e0a16d2c2a156db27e988caf46380c
